content/posts/2025-07-28-Building_Your_First_A2A_Agent.adoc

```diff
@@ -0,0 +1,432 @@
+                <artifactId>maven-war-plugin</artifactId>
+                <version>${version.war.plugin}</version>
+                <configuration>
+                    <!-- Jakarta EE doesn't require web.xml, Maven needs to catch up! -->
+                    <failOnMissingWebXml>false</failOnMissingWebXml>
+                </configuration>
+            </plugin>
+
+            <!-- The WildFly plugin deploys your war to a local JBoss AS container -->
+            <plugin>
+                <groupId>org.wildfly.plugins</groupId>
+                <artifactId>wildfly-maven-plugin</artifactId>
+                <version>${version.wildfly.maven.plugin}</version>
+                <configuration>
+                    <discoverProvisioningInfo>
+                        <spaces>
+                            <space>incubating</space>
+                        </spaces>
+                        <version>${version.wildfly.server}</version>
+                        <suggest>true</suggest>
+                    </discoverProvisioningInfo>
+                    <name>ROOT.war</name>
+                    <extraServerContentDirs>
+                        <extraServerContentDir>extra-content</extraServerContentDir>
+                    </extraServerContentDirs>
+                    <packagingScripts>
+                        <packaging-script>
+                            <scripts>
+                                <script>${basedir}/target/scripts/configure_mcp.cli</script>
+                            </scripts>
+                        </packaging-script>
+                    </packagingScripts>
+                </configuration>
+                <executions>
+                    <execution>
+                        <goals>
+                            <goal>package</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+    </build>
+</project>
+
+----
+
+[source,bash]
+----
+git clone git@github.com:ehsavoie/a2a-weather.git
+
+cd a2a-weather
+
+mvn clean package
+
+export GEMINI_CHAT_MODEL_NAME=gemini-2.0-flash-lite
+export GEMINI_API_KEY=*******************************
+
+./target/server/bin/standalone.sh -b 0.0.0.0
+----
+
+.A2A Inspector - Agent Card
+image::/assets/img/news/ai/a2a_inspector.png[A2A Agent Card]
+
+Now, if we ask for the weather in Los Angeles, California, we get the following answer:
+
+.A2A Inspector - Agent Execution
+image::/assets/img/news/ai/a2a_invokation.png[A2A Agent Execution]
+
+Our agent is now up and running and can be consumed by other agents, not just the inspector.
+
+== Conclusion
+
+As you can see, it takes very little effort to create an A2A Agent that can be easily integrated into your AI application. This agent can now communicate with different AI agents, allowing them to collaborate, share information, and coordinate actions regardless of their underlying frameworks or vendors.
```

Comment:
Maybe could reword to:

it takes very little effort to turn your AI application into an A2A agent